stripe/src/main/java/com/stripe/android/paymentsheet/repositories/PaymentMethodsRepository.kt

```diff
@@ -0,0 +1,62 @@
+package com.stripe.android.paymentsheet.repositories
+
+import com.stripe.android.model.ListPaymentMethodsParams
+import com.stripe.android.model.PaymentMethod
+import com.stripe.android.networking.ApiRequest
+import com.stripe.android.networking.StripeRepository
+import com.stripe.android.paymentsheet.PaymentSheet
+import kotlinx.coroutines.withContext
+import kotlin.coroutines.CoroutineContext
+
+internal sealed class PaymentMethodsRepository {
+    abstract suspend fun get(
+        customerConfig: PaymentSheet.CustomerConfiguration,
+        type: PaymentMethod.Type
+    ): List<PaymentMethod>
+
+    /**
+     * Retrieve the [PaymentMethod] list from a static source.
+     */
+    class Static(
+        private val paymentMethods: List<PaymentMethod>
+    ) : PaymentMethodsRepository() {
+        override suspend fun get(
+            customerConfig: PaymentSheet.CustomerConfiguration,
+            type: PaymentMethod.Type
+        ): List<PaymentMethod> = paymentMethods
+    }
+
+    /**
+     * Retrieve the [PaymentMethod] list from the API.
+     */
+    class Api(
+        private val stripeRepository: StripeRepository,
+        private val publishableKey: String,
+        private val stripeAccountId: String?,
+        private val workContext: CoroutineContext
+    ) : PaymentMethodsRepository() {
+        override suspend fun get(
+            customerConfig: PaymentSheet.CustomerConfiguration,
+            type: PaymentMethod.Type
+        ): List<PaymentMethod> = withContext(workContext) {
+            runCatching {
```

Comment:
How much control do we want the client to have? I can imagine a client that would want to handle the context and exception handling themselves to, eg, log a specific error when they're unable to fetch a payment method. 

But on the other hand, wrapping everything like this makes the API surface easier to understand. Wdyt?

Reply:
Yeah, that makes sense. This class has a pretty narrow focus so I'm not overly concerned about different use cases. I'm going to leave this as-is, but we should probably come to consensus on the right approach.